src/Generic/Random/Internal/Generic.hs

```diff
@@ -190,6 +190,9 @@ data Weights a = Weights (Weights_ (Rep a)) Int
 -- @
 -- ((9 :: 'W' \"Leaf\") '%' (8 :: 'W' \"Node\") '%' ())
 -- @
+--
+-- Note: these weights are only checked on GHC 8.0 or newer. They are ignored
```

Comment:
This sounds a bit as if the frequencies, too, were ignored. Is this clearer?

```suggestion
-- Note: these annotations are only checked on GHC 8.0 or newer. They are ignored
```

Reply:
That's indeed better, thanks!